src/gpu/intel/sycl/stream.cpp

```diff
@@ -174,6 +174,67 @@ status_t stream_t::resume_recording() {
     return status::success;
 }
 
+status_t stream_t::init_async_tracking(std::string &vinfo, double *start_ms) {
+    stream_timing_data_ = std::make_shared<async_timing_data_t>();
+    stream_timing_data_->start_ms = get_msec();
+    stream_timing_data_->profiler_enabled = is_profiling_enabled();
+    stream_timing_data_->vinfo = vinfo;
+    return status::success;
+}
+
+status_t stream_t::register_async_tracker(const ::sycl::event &event) {
+
+    if (!stream_timing_data_) return status::runtime_error;
+
+    impl()->queue()->submit([&](::sycl::handler &cgh) {
+        cgh.depends_on(event);
+        auto td = stream_timing_data_;
+        cgh.host_task([td, event]() {
+            try {
+                if (td->profiler_enabled) {
+                    auto start_ms = event.get_profiling_info<
+                            ::sycl::info::event_profiling::command_start>();
+                    auto end_ms = event.get_profiling_info<
+                            ::sycl::info::event_profiling::command_end>();
+                    td->start_ms = start_ms * 1e-6;
+                    td->end_ms = end_ms * 1e-6;
+                    td->duration_ms = (end_ms - start_ms) * 1e-6;
+                } else {
+                    td->end_ms = get_msec();
```

Comment:
I suggest to move this branch and handling out of the host task - we can check and exit earlier, without submitting a task and adding extra overhead.

Reply:
I guess you can check whether the profiling capabilities are enabled before the host task. But anyway, after thinking more, my main concern was the overhead, and since this is specifically for verbose when overhead is expected, this shouldn't be very important. Feel free to resolve.